### PR TITLE
Various fixes to be closer to the current code in Tradeshift/Apps-Server

### PR DESCRIPTION
**Pull request by sampi**
@sqren @tstapleton @nealstewart @wejendorp

### CODE REVIEW

**Comment by sqren** [comment]
If this is Apps-Server specific fixes, it should go in:
https://github.com/Tradeshift/Apps-Server/blob/master/.eslintrc.json

File: eslintrc.json [hunk 1]
```diff
@@ -27,12 +27,14 @@
 		"keyword-spacing": [2, { "before": true, "after": true }],
 		"linebreak-style": 2,
 		"no-empty": [2, { "allowEmptyCatch": true }],
-		"no-invalid-this": 2,
 		"no-loop-func": 2,
 		"no-script-url": 2,
 		"no-shadow": 2,
 		"no-tabs": 0,
+		"no-template-curly-in-string": 1,
+		"no-throw-literal": 1,
```
Comment (sqren) [inline eslintrc.json]: Any reason this shouldn't be an error?

Comment (sqren) [inline eslintrc.json]: Actually, why do we have warnings for these - shouldn't they be errors?

```
"block-scoped-var": 1,
"camelcase": [1, { "properties": "never" }],
"no-template-curly-in-string": 1,
"no-throw-literal": 1,
```

Comment (nealstewart) [inline eslintrc.json]: i'd say we should just fix the literal throws ;)

Comment (sqren) [inline eslintrc.json]: > i'd say we should just fix the literal throws ;)

That's not enough. If we want to catch these in the future, we should enforce it by breaking the build.

Comment (wiredearp) [inline eslintrc.json]: @sqren It can only ever lead to an unintentional error in ES6, until then it's just more of a future-proofing initiative I guess.

Comment (sqren) [inline eslintrc.json]: I don't know about `template-curly`. I'm talking about `no-throw-literal`

Comment (wiredearp) [inline eslintrc.json]: Actually, `no-thow-literal`should probably be an error, since it would otherwise not produce much of a strack trace.

Comment (sampi) [inline eslintrc.json]: I think we should encourage returning an `Object`, but if you only need a `String`, why not do it?

**Comment by wejendorp** [comment]
I tend to agree with @sqren, I like having this as the end-goal.
Then we can remove overrides as we improve our code in apps.

**Comment by sampi** [comment]
Everything's an error now!

**Comment by sqren** [comment]
👍

File: eslintrc.json [hunk 2]
```diff
@@ -18,21 +18,23 @@
...
 		"keyword-spacing": [2, { "before": true, "after": true }],
 		"linebreak-style": 2,
 		"no-empty": [2, { "allowEmptyCatch": true }],
-		"no-invalid-this": 2,
 		"no-loop-func": 2,
 		"no-script-url": 2,
 		"no-shadow": 2,
 		"no-tabs": 0,
+		"no-template-curly-in-string": 0,
+		"no-useless-call": 0,
```
Comment (sampi) [inline eslintrc.json]: just made another fix to re-enable it.

**Pull request merged**